### PR TITLE
ci: Updaiting workflow permissions and adding hardend runner per step-security.

### LINKED ISSUE #3121: Update default GITHUB_TOKEN permissions per step-security recommendations
**Files to change**:

- `.github/workflows/acceptance-workflow.yml`
- `.github/workflows/charts.yml`
- `.github/workflows/conformity-workflow.yml`
- `.github/workflows/dapp.yml`
- `.github/workflows/dev-tool-workflow.yml`
- `.github/workflows/flow-pr-title-check.yml`
- `.github/workflows/image-build.yml`
- `.github/workflows/image-build.yml`
- `.github/workflows/postman.yml`
- `.github/workflows/release-acceptance.yml`
- `.github/workflows/release-integration.yml`
- `.github/workflows/release-production.yml`
- `.github/workflows/subgraph.yml`

*Changes to add*:

```suggestion
# Add the following
permissions:
  contents: read
```

**Add Hardened Runner**:
`.github/workflows/pr-label-milestone-check.yml`

### PR DESCRIPTION
**Pull request by san-est**
**Description**:
Updating several workflows and their permissions per step-security requirements and adding hardend runner to one of the workflows.

**Related issue(s)**:

Fixes #3121 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]

File: .github/workflows/pr-label-milestone-check.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ jobs:
     runs-on: ubuntu-latest
```
Comment (rbarkerSL) [inline .github/workflows/pr-label-milestone-check.yml:9]: Need to update this runner to be the self-hosted runner.

```suggestion
    runs-on: smart-contracts-linux-medium
```

File: .github/workflows/conformity-workflow.yml
Comment (rbarkerSL) [inline .github/workflows/conformity-workflow.yml:1]: @Nana-EC node version 18 is end of life at the end of April 2025; can we update this to use node 20 LTS or node 22 (preferred)?

File: .github/workflows/release-acceptance.yml
Comment (rbarkerSL) [inline .github/workflows/release-acceptance.yml:1]: In the job `release-acceptance-test` the `Authenticate` step (line 31) should be modified as follows:

```suggestion
      - name: Authenticate
        env:
          HEDERA_PORTAL_USER: ${{ secrets.HEDERA_PORTAL_USER }}
          HEDERA_PORTAL_PASSWORD: ${{ secrets.HEDERA_PORTAL_PASSWORD }}
        run: |
            RESPONSE=$(jq --null-input -r --arg USER "${HEDERA_PORTAL_USER}" --arg PASS "${HEDERA_PORTAL_PASSWORD}" '{"email": $USER, "password": $PASS}' | curl -sSL -c /tmp/cookiejar.bin --data @- -X POST -H "Accept: application/json" -H "Content-Type: application/json" https://portal.hedera.com/api/session)
            echo "::add-mask::$RESPONSE"
            echo "The response was: $RESPONSE"
            ACCOUNTS_JSON="$(curl -sSL -b /tmp/cookiejar.bin -H "Accept: application/json" https://portal.hedera.com/api/account)"
            echo "::add-mask::$ACCOUNTS_JSON"
            TESTNET_ACCOUNT=$(echo $ACCOUNTS_JSON | jq -r '.accounts[] | select(.network=="testnet") | .accountNum')
            echo "The testnet account is: $TESTNET_ACCOUNT"
            PUBLIC_KEY=$(echo $ACCOUNTS_JSON | jq -r '.accounts[] | select(.network=="testnet") | .publicKey')
            echo "The publicKey is: $PUBLIC_KEY"
            PRIVATE_KEY=$(echo $ACCOUNTS_JSON | jq -r '.accounts[] | select(.network=="testnet") | .privateKey')
            echo "::add-mask::$PRIVATE_KEY"
            echo "The privateKey is: $PRIVATE_KEY"
            echo "OPERATOR_ID_MAIN=0.0.${TESTNET_ACCOUNT}" >> $GITHUB_ENV
            echo "OPERATOR_KEY_MAIN=${PRIVATE_KEY}" >> $GITHUB_ENV
```

**Review by rbarkerSL — CHANGES_REQUESTED** [review]

File: .github/workflows/release-acceptance.yml
Comment (san-est) [reply .github/workflows/release-acceptance.yml:1]: Updated in latest commit.

**Review by rbarkerSL — APPROVED** [review]

**Review by quiet-node — APPROVED** [review]

File: .github/workflows/conformity-workflow.yml
Comment (Nana-EC) [reply .github/workflows/conformity-workflow.yml:1]: Yes we can, but we'll do it in a separate PR.
LTS is preferred so we'll start there and see later about 22

Comment (Nana-EC) [reply .github/workflows/conformity-workflow.yml:1]: Captured this here - https://github.com/hashgraph/hedera-json-rpc-relay/issues/3156

**Comment by Nana-EC** [comment]
@san-est need to fix DCO then you can check in

**Pull request merged**